userspace/libsinsp/async_cgroup.cpp

```diff
@@ -0,0 +1,97 @@
+#include "async_cgroup.h"
+
+#include <fstream>
+#include "sinsp.h"
+
+namespace {
+// to prevent 32-bit number of kilobytes from overflowing, ignore values larger than 4 TiB.
+// This reports extremely large values (e.g. almost-but-not-quite 9EiB as set by k8s) as unlimited.
+// Note: we use the same maximum value for cpu shares/quotas as well; the typical values are much lower
+// and so should never exceed CGROUP_VAL_MAX either
+constexpr const int64_t CGROUP_VAL_MAX = 1UL << 42u;
+
+bool read_cgroup_val(std::shared_ptr<std::string>& subsys, const std::string& cgroup, const std::string& filename, int64_t& out)
+{
+	if(cgroup == "/")
+	{
+		return false;
+	}
+
+	std::string path = *subsys.get() + "/" + cgroup + "/" + filename;
+	std::ifstream cg_val(path);
+
+	int64_t val;
+	cg_val >> val;
```

Comment:
Major: `val` is uninitialized here, no?

Reply:
yup, we're reading from the ifstream to val. Does it need to be initialized if we overwrite it immediately? I guess I'm not 100% clear on what happens e.g. on read failure -- does val remain uninitialized or is it set to a specific value?